Source/Math/NcclComm.cpp

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) 2016, NVIDIA CORPORATION. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
```

Comment:
Ah, I see. Yes, it get's compiled whenever CUDA is enabled. I meant to do something like this:

```
ifdef NCCL_PATH
        MATH_SRC += $(SOURCEDIR)/Math/NcclComm.cpp 
endif
```
But if you prefer that I move the include guards into NcclComm.cpp that would work too.